lib/SharedLib/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+# We supply empty.cpp because CMake does not allow creating a library without sources.
+add_library(TTMLIRSO SHARED empty.cpp)
+
+set(METAL_LIB_DIR "${CMAKE_SOURCE_DIR}/third_party/tt-metal/src/tt-metal-build/lib")
```

Comment:
This should be replaced with `TTMETAL_LIBRARY_DIR`, which is exported from `third_party/CMakeLists.txt`.